mitosheet/src/streamlit/MitoStreamlitWrapper.tsx

```diff
@@ -160,10 +169,64 @@ class MitoStreamlitWrapper extends StreamlitComponentBase<State> {
         const response = await this.getResponseData(msg['id'] as string);        
         return response;
     }
+
+    componentWillUnmount() {
+        if (this.processMessageQueueTimer) {
+            clearTimeout(this.processMessageQueueTimer);
+        }
+    }
+
+    processSetComponentValueQueue = () => {
+        if (this.state.setComponentValueQueue.length > 0) {
+            // Send one message
+            const value = this.state.setComponentValueQueue[0];
+            Streamlit.setComponentValue(value);
+
+            // Remove the processed message from the queue - making sure
+            // to avoid race conditons by finding by value
+            this.setState((prevState) => {
+                const valueQueue = [...prevState.setComponentValueQueue];
+                const index = valueQueue.findIndex((m) => m === value);
+                valueQueue.splice(index, 1);
+
+                return { 
+                    setComponentValueQueue: valueQueue,
+                    isSettingComponentValue: valueQueue.length > 0,
+                };
+            });
+
+            // Set a timer to process the next message after a delay
+            this.processMessageQueueTimer = setTimeout(this.processSetComponentValueQueue, DELAY_BETWEEN_SET_COMPONENT_VALUES);
+        } else {
+            // Otherwise, we have processed the full queue
+            this.setState({ isSettingComponentValue: false });
+        }
+    };
+
+    setComponentValue = (value: any) => {
```

Comment:
One thing I'm wondering -- can we drop some of the selections? I think yes. I'm going to get rid of all this debouncing infrastructure.